gqcpy/src/QCMethod/CC/CCD_bindings.cpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "Mathematical/Algorithm/IterativeAlgorithm.hpp"
+#include "QCMethod/CC/CCD.hpp"
+#include "QCMethod/CC/CCDSolver.hpp"
+
+#include <pybind11/pybind11.h>
+
+
+namespace py = pybind11;
+
+
+namespace gqcpy {
+
+
+void bindQCMethodCCD(py::module& module) {
+    py::class_<GQCP::QCMethod::CCD<double>>(module, "CCD", "The CCD quantum chemical method.")
+
+        .def_static(
+            "optimize",
+            [](GQCP::IterativeAlgorithm<GQCP::CCSDEnvironment<double>>& solver, GQCP::CCSDEnvironment<double>& environment) {
+                return GQCP::QCMethod::CCD<double>().optimize(solver, environment);
+            },
+            "Optimize the CCD wave function model.");
+}
```

Comment:
Could you add the relevant `py::arg`s here too?

Reply:
Aren't the `py::arg`s the right ones? Since we use `CCSDEnvironment` for our CCD implementation.

Reply:
The parameters for the lambda function are indeed correct, but could you add `py::arg`s to the `.def_static()` method accordingly?

Reply:
I don't see what's wrong with the arguments of `.def_static()`. I've used the same args as in `GCMethod/CC/CCSD_bindings.cpp`, since both methods have a similar functionality.

Reply:
@dariatols There is nothing 'wrong' with what you've written, rather, it's incomplete: you are not providing `py::arg` arguments. 

You're providing `"optimize"` (which is the function name), the lambda, and `"Optimize the CCD wave function model."`, but I don't see any `py::arg` calls. Take a look to the other files in the Python bindings to check what you're missing. If `QCMethod/CC/CCSD_bindings.cpp` is also missing these `py::arg` calls, please add them accordingly.

Reply:
I've changed `QCMethod/CC/CCD_bindings.cpp` with `py::arg`s as an example. I'll do the rest after my exam.